### PR TITLE
[CELEBORN-748] Rename RssHARetryClient to MasterClient

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?

Rename RssHARetryClient to MasterClient

### Why are the changes needed?

Code refactor

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass GA.

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @pan3793 @waitinfuture @FMX

File: common/src/main/java/org/apache/celeborn/common/client/MasterClientWithRetry.java [hunk 1]
```diff
@@ -48,8 +48,8 @@
 import org.apache.celeborn.common.rpc.RpcTimeout;
 import org.apache.celeborn.common.util.ThreadUtils;
 
-public class RssHARetryClient {
-  private static final Logger LOG = LoggerFactory.getLogger(RssHARetryClient.class);
+public class MasterClientWithRetry {
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/client/MasterClientWithRetry.java:51]: No strong option, but I would call it `MasterClient` directly. What do you think? @waitinfuture @FMX

Comment (RexXiong) [reply common/src/main/java/org/apache/celeborn/common/client/MasterClientWithRetry.java:51]: > No strong opinion, but I would call it `MasterClient` directly. What do you think? @waitinfuture @FMX

+1

Comment (AngersZhuuuu) [reply common/src/main/java/org/apache/celeborn/common/client/MasterClientWithRetry.java]: Done

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to main/0.3

**Pull request closed without merging**